### DISCUSSION TITLE
Can we have (almost) empty macOS runners?

### DISCUSSION THREAD

**Discussion by ringods** [actions/virtual-environments#5500] [Image configuration]
### Tool name

No tools

### Tool license

Not applicable

### Add or update?

- [ ] Add
- [ ] Update

### Desired version

Not applicable

### Brief description of tool

Thanks for having macOS runners.

Besides doing software builds, where Github Actions is mainly targeted to, I also leverage the macOS runners to verify the automation that manages my MBP: https://github.com/ringods/macos-config
Similar project: https://github.com/geerlingguy/mac-dev-playbook

The macOS runners come preinstalled with a lot of tools, which for regular software builds, are a prerequisite. For the validation of my automated setup, I would like to start from an (almost) empty VM.

Would it be possible to have variants of the macOS runner setup without all the pre-installed tools?

### Provide a basic test case to validate the tool's functionality.

```bash
* Start with an empty VM
* In my GHA setup, I install Homebrew & Ansible
* I then run my automated macOS setup
* I validate everything works.
```

### Virtual environments affected

- [ ] Ubuntu 18.04
- [ ] Ubuntu 20.04
- [ ] Ubuntu 22.04
- [X] macOS 10.15
- [X] macOS 11
- [X] macOS 12
- [ ] Windows Server 2019
- [ ] Windows Server 2022

**Comment by al-cheb** [comment]
Hello @ringods ,
Currently, we don't have any plans to provide clean images. You should create your own macOS self-hosted agents.

**Comment by samus-aran** [comment]
We're locking this discussion because it has not had recent activity and/or other members have asked for more information to assist you but received no response. Thank you for helping us maintain a productive and tidy community for all our members.
